lib/logging/log.js

```diff
@@ -0,0 +1,531 @@
+ * @private
+ */
+var ServiceObject = require('../common/service-object.js');
+
+/**
+ * A log is a named collection of entries, each entry representing a timestamped
+ * event. Logs can be produced by Google Cloud Platform services, by third-party
+ * services, or by your applications. For example, the log `apache-access` is
+ * produced by the Apache Web Server, but the log
+ * `compute.googleapis.com/activity_log` is produced by Google Compute Engine.
+ *
+ * @resource [Introduction to Logs]{@link https://cloud.google.com/logging/docs/api/#logs}
+ *
+ * @alias module:logging/log
+ * @constructor
+ *
+ * @param {object} options - [Configuration object](#/docs/?method=gcloud).
+ *
+ * @example
+ * var gcloud = require('gcloud')({
+ *   keyFilename: '/path/to/keyfile.json',
+ *   projectId: 'grape-spaceship-123'
+ * });
+ *
+ * var logging = gcloud.logging();
+ * var log = logging.log('my-log');
+ */
+function Log(logging, name) {
+  var methods = {
+    /**
+     * Delete the log.
+     *
+     * @resource [projects.logs.delete API Documentation]{@link https://cloud.google.com/logging/docs/api/ref/rest/v2beta1/projects.logs/delete}
+     *
+     * @param {function=} callback - The callback function.
+     * @param {?error} callback.err - An error returned while making this
+     *     request.
+     * @param {object} callback.apiResponse - The full API response.
+     *
+     * @example
+     * log.delete(function(err, apiResponse) {
+     *   if (!err) {
+     *     // The log was deleted.
+     *   }
+     * });
+     */
+    delete: true
+  };
+
+  this.name = name.replace(/^.*logs\//, '');
+  this.formattedName_ = Log.formatName_(logging.projectId, name);
+
+  ServiceObject.call(this, {
+    parent: logging,
+    baseUrl: '/logs',
+    id: this.name,
+    methods: methods
+  });
+}
+
+nodeutil.inherits(Log, ServiceObject);
+
+/**
+ * Return an array of log entries with the desired severity assigned.
+ *
+ * @private
+ *
+ * @param {object|object[]} entries - Log entries.
+ * @param {string} severity - The desired severity level.
+ */
+Log.assignSeverityToEntries_ = function(entries, severity) {
+  var entriesClone = [].slice.call(entries);
+  return arrify(entriesClone).map(propAssign('severity', severity));
+};
+
+/**
+ * Format the name of a log. A log's full name is in the format of
+ * 'projects/{projectId}/logs/{logName}'.
+ *
+ * @private
+ *
+ * @return {string}
+ */
+Log.formatName_ = function(projectId, name) {
+  if (name.indexOf('logs/') > -1) {
+    return name;
+  }
+
+  var uriSafeName = name;
+
+  if (decodeURIComponent(name) === name) {
+    // The name has not been encoded yet.
+    uriSafeName = encodeURIComponent(name);
+  }
+
+  return format('projects/{projectId}/logs/{name}', {
+    projectId: projectId,
+    name: uriSafeName
+  });
+};
+
+/**
+ * Write a log entry with a severity of "ALERT".
+ *
+ * This is a simple wrapper around {module:logging/log#insert}. All arguments
+ * are the same as documented there.
```

Comment:
Could we just do a `@mixin` here instead? It might be a little tedious to jump from method to method? Especially if the method has been deep linked into.